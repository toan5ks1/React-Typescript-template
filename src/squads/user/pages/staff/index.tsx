import StaffList from "src/squads/user/modules/staff-list";

import WrapperPageContent from "src/components/Wrappers/WrapperPageContent";
import TranslationProvider from "src/squads/user/providers/TranslationProvider";

export default function StaffListPage() {
    return (
        <TranslationProvider>
            <WrapperPageContent>
                <StaffList />
            </WrapperPageContent>
        </TranslationProvider>
    );
}
