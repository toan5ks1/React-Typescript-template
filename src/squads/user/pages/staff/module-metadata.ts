import { Entities, Features } from "src/common/constants/enum";
import { IAppResource } from "src/models/resource";
import { MicroFrontendTypes } from "src/routing/type";

import { SchoolOutlined } from "@mui/icons-material";

const staff: IAppResource = {
    key: Entities.STAFF,
    name: Entities.STAFF,
    translateKey: "name",
    basename: `/${MicroFrontendTypes.USER}`,
    icon: SchoolOutlined,

    feature: Features.STAFF_MANAGEMENT,
};
export default staff;
