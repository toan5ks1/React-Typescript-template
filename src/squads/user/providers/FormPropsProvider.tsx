import { createContext, PropsWithChildren, useContext } from "react";

export interface FormPropsContextValue {
    readOnly: boolean;
}

const FormPropsContext = createContext<FormPropsContextValue>({ readOnly: false });

const { Provider } = FormPropsContext;

export const FormPropsProvider = ({
    children,
    readOnly,
}: PropsWithChildren<{ readOnly: boolean }>) => {
    return <Provider value={{ readOnly }}>{children}</Provider>;
};

export default FormPropsContext;
export const useFormProps = () => useContext(FormPropsContext);
