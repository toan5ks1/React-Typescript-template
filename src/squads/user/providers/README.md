### We abstract the provider into an isolated component, can be hard to test but parent component don't need to care about what the provider does
