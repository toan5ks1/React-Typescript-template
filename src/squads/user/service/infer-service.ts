import standaloneQueryClient from "src/squads/user/internals/standalone-query-client";
import brightcoveServiceYasuo from "src/squads/user/service/yasuo/brightcove-yasuo-service";

import classService from "./define-service/class-service";
import courseAccessPathService from "./define-service/course-access-paths-service";
import courseService from "./define-service/course-service";
import courseStudentServiceEureka from "./define-service/course-student-service-eureka";
import grantedRoleAccessPathService from "./define-service/granted-role-access-path-service";
import grantedRoleService from "./define-service/granted-role-service";
import locationService from "./define-service/location-service";
import organizationsService from "./define-service/organizations-service";
import parentImportService from "./define-service/parent-import-service";
import parentService from "./define-service/parents-service";
import passwordFirebaseService from "./define-service/password-firebase-service";
import prefectureService from "./define-service/prefecture-service";
import roleService from "./define-service/role-service";
import schoolCourseService from "./define-service/school-course-service";
import schoolInfoService from "./define-service/school-info-service";
import schoolLevelService from "./define-service/school-level-service";
import staffService from "./define-service/staff-service";
import studentEntryExitService from "./define-service/student-entry-exit-service";
import studentImportService from "./define-service/student-import-service";
import studentPackageClassService from "./define-service/student-package-class-service";
import studentPackageService from "./define-service/student-package-service";
import studentParentService from "./define-service/student-parents-service";
import studentService from "./define-service/student-service";
import userAccessPathService from "./define-service/user-access-paths-service";
import userAddressService from "./define-service/user-address-service";
import userGroupService from "./define-service/user-group-service";
import userService from "./define-service/user-service";
import {
    composeServices,
    createUseQuery,
    createUseMutation,
    createUseQueryPagination,
} from "./service-creator";

// compose all services into service map
const rootService = composeServices({
    brightcove: brightcoveServiceYasuo,
    studentParent: studentParentService,
    student: studentService,
    parent: parentService,
    users: userService,
    course: courseService,
    userAccessPath: userAccessPathService,
    studentPackage: studentPackageService,
    courseStudents: courseStudentServiceEureka,
    location: locationService,
    studentEntryExit: studentEntryExitService,
    userGroup: userGroupService,
    staff: staffService,
    studentImport: studentImportService,
    grantedRole: grantedRoleService,
    grantedRoleAccessPath: grantedRoleAccessPathService,
    studentPackageClass: studentPackageClassService,
    class: classService,
    role: roleService,
    parentImport: parentImportService,
    courseAccessPath: courseAccessPathService,
    organizations: organizationsService,
    passwordFirebase: passwordFirebaseService,
    schoolCourse: schoolCourseService,
    schoolInfo: schoolInfoService,
    schoolLevel: schoolLevelService,
    userAddress: userAddressService,
    prefecture: prefectureService,
});

// infer typing for useQuery, we must use another infer layer because of typescript limitation
// else we will need to pass full generic on usage useQuery<A, B, C, D, E>().
export const inferQuery = createUseQuery(rootService);

export const inferQueryPagination = createUseQueryPagination(rootService);

export const inferMutation = createUseMutation(rootService);

export const inferStandaloneQuery = standaloneQueryClient.createStandaloneQuery(rootService);
