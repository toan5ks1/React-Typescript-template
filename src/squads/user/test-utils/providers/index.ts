export { default as TestThemeProvider } from "./TestThemeProvider";
export { default as TestCommonAppProvider } from "./TestCommonAppProvider";
export * from "./TestQueryWrapper";
export { default as TestHookForm } from "./TestHookForm";
export { default as TestHookFormProvider } from "./TestHookFormProvider";
export { default as TestingRouter } from "./TestingRouter";
export { default as AppProvider } from "./AppProvider";
export { default as TestStudentDetailProvider } from "./TestStudentDetailProvider";
