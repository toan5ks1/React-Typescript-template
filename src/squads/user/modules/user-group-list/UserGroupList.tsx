import { Entities, ModeOpenDialog } from "src/common/constants/enum";
import { UserGroupUpsert } from "src/squads/user/modules/user-group-upsert";
import { inferQueryPagination } from "src/squads/user/service/infer-service";

import Box from "@mui/material/Box";
import ButtonCreate from "src/components/Buttons/ButtonCreate";
import TypographyPageTitle from "src/components/Typographys/TypographyPageTitle";
import WrapperPageContent from "src/components/Wrappers/WrapperPageContent";
import UserGroupTable from "src/squads/user/modules/user-group-list/components/UserGroupTable";

import useDialog from "src/squads/user/hooks/useDialog";
import useResourceTranslate from "src/squads/user/hooks/useResourceTranslate";
import useShowSnackbar from "src/squads/user/hooks/useShowSnackbar";
import useTranslate from "src/squads/user/hooks/useTranslate";

const UserGroupList = () => {
    const t = useTranslate();
    const tUserGroup = useResourceTranslate(Entities.USER_GROUP);
    const showSnackbar = useShowSnackbar();
    const { open, onOpen, onClose } = useDialog();
    const {
        result: { isFetching, refetch },
        data,
        pagination: userGroupPagination,
    } = inferQueryPagination({
        entity: "userGroup",
        action: "userGetManyUserGroupsWithFilter",
    })(
        {},
        {
            enabled: true,
            onError: (error) => {
                window.warner?.warn(`userGetManyUserGroupsWithFilter`, error);
                showSnackbar(t("ra.message.unableToLoadData"), "error");
            },
        }
    );

    return (
        <WrapperPageContent>
            <TypographyPageTitle title={tUserGroup("title")} />
            <Box display="flex" justifyContent="flex-end" alignItems="start" mb={2}>
                <ButtonCreate
                    data-testid="UserGroupTable__addButton"
                    resource={Entities.USER_GROUP}
                    onClick={onOpen}
                >
                    {t("ra.common.action.add")}
                </ButtonCreate>
            </Box>
            <UserGroupTable
                data={data?.data || []}
                isFetching={isFetching}
                pagination={userGroupPagination}
            />
            {open ? (
                <UserGroupUpsert
                    open={open}
                    mode={ModeOpenDialog.ADD}
                    onClose={onClose}
                    onSuccess={refetch}
                />
            ) : null}
        </WrapperPageContent>
    );
};

export default UserGroupList;
