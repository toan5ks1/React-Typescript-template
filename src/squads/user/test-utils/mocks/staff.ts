import { ArrayElement } from "src/common/constants/types";
import {
    User_StaffListV2Query,
    User_StaffListV3Query,
    User_StaffOneV2Query,
    User_StaffListV4Query,
    User_UserGroupsManyReferenceV2Query,
} from "src/squads/user/service/bob/bob-types";
import { Staff, Teacher as TeacherOne } from "src/squads/user/service/bob/user-service-bob/types";
import { PaginationWithTotal } from "src/squads/user/service/service-creator";
import { userGroupsListToUserGroupIdsList } from "src/squads/user/service/usermgmt/staff-service-usermgmt";
import { NsUsermgmtStaffService } from "src/squads/user/service/usermgmt/staff-service-usermgmt/types";
import { CountryKeys } from "src/squads/user/typings/remote";

import { Country } from "manabuf/common/v1/enums_pb";
import { UserGroup } from "manabuf/usermgmt/v2/enums_pb";
import {
    CreateStaffResponse,
    UpdateStaffResponse,
    UpdateStaffSettingResponse,
} from "manabuf/usermgmt/v2/users_pb";

export const MOCK_STAFF_USERS_DATA = {
    name: "Thomas",
    email: "thomas@manabie.com",
    user_group: "USER_GROUP_TEACHER",
    avatar: "",
    country: "COUNTRY_JP",
    phone_number: "",
};
export interface Pagination {
    count: number;
    onPageChange: () => void;
    page: number;
    rowsPerPage: number;
    limit: number;
    offset: number;
    onRowsPerPageChange: () => void;
}

export interface TeacherInformation {
    teacher_by_school_ids: TeacherOne["teacher_by_school_ids"];
    users: TeacherOne["users"];
    teacher_id: TeacherOne["teacher_id"];
    id: TeacherOne["teacher_id"];
    country: CountryKeys;
}

export const mockPagination: Pagination = {
    count: 1,
    onPageChange: jest.fn(),
    page: 0,
    rowsPerPage: 10,
    limit: 10,
    offset: 0,
    onRowsPerPageChange: jest.fn(),
};

export const mockListStaffWithFilterV2: User_StaffListV2Query = {
    find_teacher_by_school_id: [...Array(5).keys()].map((value) => ({
        name: `Teacher ${value}`,
        email: `teach_email_${value}@gmail.com`,
        user_id: `user_id_${value}`,
        resource_path: "-2147483644",
    })),
    find_teacher_by_school_id_aggregate: {
        aggregate: {
            count: 5,
        },
    },
};

export const mockListStaffWithFilterV3: User_StaffListV3Query = {
    staff: [...Array(5).keys()].map((value) => ({
        staff_id: `staff_id_${value}`,
        user: {
            name: `Staff Name ${value}`,
            email: `staff_name_${value}@gmail.com`,
            user_group_members: [
                {
                    user_group: {
                        user_group_name: `Teacher ${value}`,
                        user_group_id: `user_group_id_${value}`,
                    },
                },
            ],
        },
    })),
    staff_aggregate: {
        aggregate: {
            count: 5,
        },
    },
};

export const mockListStaffWithFilterV4: User_StaffListV4Query = {
    staff: [...Array(5).keys()].map((value) => ({
        staff_id: `staff_id_${value}`,
        user: {
            name: `Staff Name ${value}`,
            email: `staff_name_${value}@gmail.com`,
            resource_path: "-2147483644",
            user_group_members: [
                {
                    user_group: {
                        user_group_name: `Teacher ${value}`,
                        user_group_id: `user_group_id_${value}`,
                    },
                },
            ],
        },
    })),
    staff_aggregate: {
        aggregate: {
            count: 5,
        },
    },
};

type StaffListQueryReturn = {
    data: User_StaffListV4Query["staff"];
    total: number;
};

export const mockListStaffReturnV4: StaffListQueryReturn = {
    data: [...Array(5).keys()].map((value) => ({
        staff_id: `staff_id_${value}`,
        user: {
            name: `Staff Name ${value}`,
            email: `staff_name_${value}@gmail.com`,
            resource_path: "-2147483644",
            user_group_members: [...Array(5).keys()].map((val) => ({
                user_group: {
                    user_group_name: `Teacher ${val}`,
                    user_group_id: `user_group_id_${val}`,
                },
            })),
        },
    })),
    total: 5,
};
interface StaffTableProp {
    data: User_StaffListV4Query["staff"];
    pagination: PaginationWithTotal;
    isFetching: boolean;
}

export const staffTableProp: StaffTableProp = {
    data: mockListStaffReturnV4.data,
    isFetching: false,
    pagination: mockPagination,
};

export const mockReturnStaffData = {
    data: mockListStaffWithFilterV4.staff,
    total: 1,
};

export const staffOneMockData: ArrayElement<User_StaffOneV2Query["staff"]> = {
    staff_id: "staff-id",
    user: {
        user_id: "Sierra",
        name: "Sierra",
        email: "sierra2022@gmail.com",
        user_group: "USER_GROUP_TEACHER",
        country: "COUNTRY_JP",
        user_group_members: [
            {
                user_group: {
                    user_group_id: "user_group_01",
                    user_group_name: "Student A",
                },
            },
            {
                user_group: {
                    user_group_id: "user_group_02",
                    user_group_name: "Student A",
                },
            },
        ],
    },
};

interface userStaffGetOneReturn {
    data: {
        staff: User_StaffOneV2Query["staff"];
    };
}

export interface StaffInformation {
    staff_id: Staff["staff_id"];
    user: ArrayElement<User_StaffOneV2Query["staff"]>["user"];
    country: CountryKeys;
}

export const mockUserStaffGetOneReturn: userStaffGetOneReturn = {
    data: {
        staff: [staffOneMockData],
    },
};

export const createMockStaff = (): StaffInformation => {
    return {
        staff_id: "staff-id",
        user: staffOneMockData.user,
        country: "COUNTRY_JP",
    };
};

export const userGroupsManyReference: User_UserGroupsManyReferenceV2Query["user_group"] = [
    ...Array(5).keys(),
].map((val) => ({
    user_group_name: `Teacher ${val}`,
    user_group_id: `user_group_id_${val}`,
}));

export const mockUseStaffInfoRulesReturn = {
    required: {
        value: true,
        message: "This field is required",
    },
    pattern: {
        email: {
            value: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
            message: "Email address is not valid",
        },
    },
    validate: {
        email: jest.fn(),
    },
};

export const createStaffReq: NsUsermgmtStaffService.CreateStaffReq = {
    name: MOCK_STAFF_USERS_DATA.name,
    email: MOCK_STAFF_USERS_DATA.email,
    avatar: MOCK_STAFF_USERS_DATA.avatar,
    country: Country[MOCK_STAFF_USERS_DATA.country],
    phoneNumber: "",
    userGroup: UserGroup[MOCK_STAFF_USERS_DATA.user_group],
    organizationId: "1",
    userGroupsList: [...Array(5).keys()].map((val) => ({
        user_group_name: `Teacher ${val}`,
        user_group_id: `user_group_id_${val}`,
    })),
};

export const createStaffResp: NsUsermgmtStaffService.CreateStaffResp = {
    staff: {
        staffId: "staff_id",
        ...createStaffReq,
        userGroupIdsList: userGroupsListToUserGroupIdsList(createStaffReq.userGroupsList),
    },
};

export const updateStaffReq: NsUsermgmtStaffService.UpdateStaffReq = {
    staffId: "staff_id",
    name: MOCK_STAFF_USERS_DATA.name + " edited",
    email: MOCK_STAFF_USERS_DATA.email,
    userGroupsList: [],
};

export const updateStaffSettingReq = (
    autoCreateTimesheet: boolean
): NsUsermgmtStaffService.UpdateStaffSettingReq => {
    return {
        staffId: "staff_id",
        autoCreateTimesheet,
    };
};

export const createMockCreateStaffResponse = (): CreateStaffResponse => {
    const staffResp = new CreateStaffResponse.StaffProfile();
    staffResp.setStaffId("staff_id");
    staffResp.setName(MOCK_STAFF_USERS_DATA.name);
    staffResp.setEmail(MOCK_STAFF_USERS_DATA.email);
    staffResp.setAvatar(MOCK_STAFF_USERS_DATA.avatar);
    staffResp.setCountry(Country[MOCK_STAFF_USERS_DATA.country]);
    staffResp.setPhoneNumber("");
    staffResp.setUserGroup(UserGroup[MOCK_STAFF_USERS_DATA.user_group]);
    staffResp.setOrganizationId("1");

    const resp = new CreateStaffResponse();
    resp.setStaff(staffResp);
    return resp;
};

export const createMockUpdateStaffResponse = (): UpdateStaffResponse => {
    const resp = new UpdateStaffResponse();
    resp.setSuccessful(true);
    return resp;
};

export const createMockUpdateStaffSettingResponse = (): UpdateStaffSettingResponse => {
    const resp = new UpdateStaffSettingResponse();
    resp.setSuccessful(true);
    return resp;
};

export const createMockTeacherUsersData = (overrides: Partial<TeacherInformation["users"]>) => {
    return {
        ...MOCK_STAFF_USERS_DATA,
        ...overrides,
    };
};

export const createMockTeacher = (): TeacherInformation => {
    return {
        id: "teacher_id",
        teacher_id: "teacher_id",
        country: "COUNTRY_JP",
        users: MOCK_STAFF_USERS_DATA,
        teacher_by_school_ids: [
            {
                school_id: 1,
            },
        ],
    };
};
