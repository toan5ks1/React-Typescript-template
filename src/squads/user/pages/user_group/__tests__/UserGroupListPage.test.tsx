import userGroupList from "src/squads/user/modules/user-group-list";
import { inferQueryPagination } from "src/squads/user/service/infer-service";
import { mockPagination } from "src/squads/user/test-utils/mocks/staff";
import { TestCommonAppProvider, TestQueryWrapper } from "src/squads/user/test-utils/providers";

import UserGroupList from "../index";

import { render, screen } from "@testing-library/react";

jest.mock("src/squads/user/hooks/useStaffFeatureFlag");

jest.mock("@mui/material/utils/useId", () => ({
    __esModule: true,
    default: () => "mui-test-id",
}));

jest.mock("src/squads/user/service/infer-service", () => {
    const original = jest.requireActual("src/squads/user/service/infer-service");
    return {
        __esModule: true,
        ...original,
        inferQueryPagination: jest.fn(),
    };
});

describe("<List /> UserGroup table", () => {
    it("should render correct UserGroups Page", () => {
        (inferQueryPagination as jest.Mock).mockReturnValue(() => ({
            result: {
                isLoading: false,
                refetch: jest.fn(),
            },
            data: { data: userGroupList },
            pagination: mockPagination,
        }));

        const { container } = render(
            <TestCommonAppProvider>
                <TestQueryWrapper>
                    <UserGroupList />
                </TestQueryWrapper>
            </TestCommonAppProvider>
        );

        expect(screen.queryByText("User Group")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
