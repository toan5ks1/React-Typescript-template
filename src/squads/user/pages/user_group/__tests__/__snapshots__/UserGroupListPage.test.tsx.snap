// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> UserGroup table should render correct UserGroups Page 1`] = `
.emotion-0 {
  margin: 0 auto;
  width: 100%;
  max-width: 1600px;
}

.emotion-1 {
  padding-bottom: 24px;
}

.emotion-2 {
  margin: 0;
  font-size: 1.25rem;
  font-weight: 500;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.6;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  margin-bottom: 16px;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 0.875rem;
  font-weight: 500;
  letter-spacing: 0.4px;
  text-transform: capitalize;
  font-family: Roboto,sans-serif;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #2196F3;
  box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2),0px 2px 2px rgba(0, 0, 0, 0.14),0px 1px 5px rgba(0, 0, 0, 0.12);
  border-radius: 4px;
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #0B79D0;
  box-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px rgba(0, 0, 0, 0.14),0px 1px 10px rgba(0, 0, 0, 0.12);
}

@media (hover: none) {
  .emotion-4:hover {
    background-color: #2196F3;
  }
}

.emotion-4:active {
  box-shadow: 0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12);
}

.emotion-4.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px rgba(0, 0, 0, 0.14),0px 1px 18px rgba(0, 0, 0, 0.12);
}

.emotion-4.Mui-disabled {
  color: #BDBDBD;
  box-shadow: none;
  background-color: #E0E0E0;
}

.emotion-5 {
  display: inherit;
  margin-right: 8px;
  margin-left: -4px;
}

.emotion-5>*:nth-of-type(1) {
  font-size: 20px;
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-7 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 600px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.emotion-9 {
  display: block;
}

.emotion-10 {
  margin-bottom: 8px;
  margin-top: 16px;
}

.emotion-11 {
  margin: 0;
  font-size: 1rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.5;
}

.emotion-12 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
  color: #9E9E9E;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <h6
        aria-label="title"
        class="MuiTypography-root MuiTypography-h6 emotion-2"
      >
        User Group
      </h6>
    </div>
    <div
      class="MuiBox-root emotion-3"
    >
      <button
        aria-label="ra.common.action.create"
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  emotion-4"
        data-testid="UserGroupTable__addButton"
        resource="user_group"
        style="word-break: keep-all;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium emotion-5"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        Add
        <span
          class="MuiTouchRipple-root emotion-7"
        />
      </button>
    </div>
    <div
      class="MuiBox-root emotion-8"
    >
      <div
        class="MuiBox-root emotion-9"
      >
        <svg
          data-testid="LookingFor__icon"
          fill="none"
          height="200"
          width="228"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#fff"
            height="153.117"
            rx="11.822"
            transform="matrix(.86765 -.49716 .50284 .86437 .196 59.058)"
            width="118.791"
          />
          <rect
            height="150.753"
            rx="10.64"
            stroke="#000"
            stroke-opacity="0.12"
            stroke-width="2.364"
            transform="matrix(.86765 -.49716 .50284 .86437 .192 59.922)"
            width="116.427"
            x="1.62"
            y="0.434"
          />
          <rect
            fill="#DEDEDE"
            height="10.805"
            rx="4.729"
            transform="matrix(.86746 -.49748 .50251 .86456 29.84 74.765)"
            width="83.315"
          />
          <rect
            fill="#DEDEDE"
            height="10.805"
            rx="4.729"
            transform="matrix(.86746 -.49748 .50251 .86456 41.168 94.252)"
            width="83.315"
          />
          <rect
            fill="#DEDEDE"
            height="10.805"
            rx="4.729"
            transform="matrix(.86746 -.49748 .50251 .86456 52.497 113.742)"
            width="62.889"
          />
          <rect
            fill="#fff"
            height="167.33"
            rx="11.822"
            transform="translate(72.003 2.99) skewX(-.031)"
            width="130.212"
          />
          <rect
            height="164.965"
            rx="10.64"
            stroke="#000"
            stroke-opacity="0.12"
            stroke-width="2.364"
            transform="translate(72.003 2.991) skewX(-.031)"
            width="127.847"
            x="1.182"
            y="1.182"
          />
          <rect
            fill="#BDBDBD"
            height="11.808"
            rx="4.729"
            transform="translate(91.44 33.992) skewX(-.032)"
            width="91.325"
          />
          <rect
            fill="#BDBDBD"
            height="11.808"
            rx="4.729"
            transform="translate(91.432 58.627) skewX(-.032)"
            width="91.325"
          />
          <rect
            fill="#BDBDBD"
            height="11.808"
            rx="4.729"
            transform="translate(91.425 83.262) skewX(-.032)"
            width="39.476"
          />
          <path
            d="M141.656 72.136c-25.094 5.5-40.977 30.162-35.445 55.106 5.532 24.947 30.38 40.684 55.476 35.183 8.792-1.927 16.46-6.211 22.468-12.035l28.127 19.647c3.948 2.757 9.404 1.683 11.997-2.349 2.441-3.792 1.446-8.83-2.267-11.424l-28.079-19.613a45.765 45.765 0 0 0 3.199-29.332c-5.533-24.946-30.38-40.684-55.476-35.183Zm-21.204 51.992c-3.793-17.101 7.096-34.049 24.352-37.831 17.253-3.782 34.295 7.038 38.087 24.136 3.792 17.101-7.096 34.049-24.352 37.831-17.253 3.782-34.295-7.038-38.087-24.136Z"
            fill="#9E9E9E"
            stroke="#fff"
            stroke-width="4.729"
          />
          <ellipse
            fill="#C4C4C4"
            opacity="0.2"
            rx="13.625"
            ry="6.216"
            transform="translate(149.382 93.729) skewX(-.032)"
          />
          <ellipse
            fill="#C4C4C4"
            opacity="0.2"
            rx="7.66"
            ry="7.347"
            transform="translate(136.488 136.521) skewX(-.032)"
          />
        </svg>
        <div
          class="MuiBox-root emotion-10"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-11"
          >
            No Result
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-12"
        >
          Please use try again with different keywords or filters
        </p>
      </div>
    </div>
  </div>
</div>
`;
